### PR TITLE
Expose ReadBufCursor::remaining and ReadBufCursor::put_slice

### PR DESCRIPTION
**Pull request by nox**
There is currently no way to write to ReadBufCursor without unsafe code, even though writing a slice to it like one would do to a Buf shouldn't require unsafe code.

### CODE REVIEW

**Comment by nox** [comment]
There is also #3515 that is similar, but more involved.

File: src/rt/io.rs [hunk 1]
```diff
@@ -244,19 +244,27 @@ impl<'data> ReadBufCursor<'data> {
         self.buf.init = self.buf.filled.max(self.buf.init);
     }
 
+    /// Returns the number of bytes that can be written from the current
+    /// position until the end of the buffer is reached.
+    ///
+    /// This value is equal tho the length of the slice returned by `as_mut()``.
```
Comment (seanmonstar) [inline src/rt/io.rs:250]:
```suggestion
    /// This value is equal to the length of the slice returned by `as_mut()``.
```

File: src/rt/io.rs [hunk 2]
```diff
@@ -244,19 +244,27 @@ impl<'data> ReadBufCursor<'data> {
...
     #[inline]
-    pub(crate) fn remaining(&self) -> usize {
+    pub fn remaining(&self) -> usize {
         self.buf.remaining()
     }
 
+    /// Transfer bytes into `self`` from `src` and advance the cursor
+    /// by the number of bytes written.
+    ///
+    /// `self` must have enough remaining capacity to contain all of `src`.
```
Comment (seanmonstar) [inline src/rt/io.rs:259]: How about adding a `# Panics` heading in between these two lines?

**Review by seanmonstar — CHANGES_REQUESTED** [review]
Seems fair to add these :)

**Comment by nox** [comment]
@seanmonstar I addressed both comments.

**Comment by nox** [comment]
Can we release 1.5 to get this please?

**Pull request merged**